snap/snapcraft.yaml

```diff
@@ -211,44 +208,41 @@ parts:
     source-checksum: sha256/b1a0915dbc91b979d06df1977fe594c3fa9b189f1f3d38743a2948c9f7634813
 
   redis-customizations:
-    plugin: copy
-    files:
-      src/redis/config/*: config/redis/
-      src/redis/scripts/*: bin/
-      src/redis/utilities/*: utilities/
+    plugin: dump
+    source: src/redis/
+    organize:
+      config/*: config/redis/
 
   # Copy over our PHP configuration file.
   php-customizations:
-    plugin: copy
-    files:
-      src/php/config/*: config/php/
-      src/php/scripts/*: bin/
-      src/php/utilities/*: utilities/
+    plugin: dump
+    source: src/php/
+    organize:
+      config/*: config/php/
     stage-packages: [mawk]
 
   # Copy over our Nextcloud configuration files
   nextcloud-customizations:
-    plugin: copy
-    files:
-      src/nextcloud/config/*: htdocs/config/
-      src/nextcloud/scripts/*: bin/
-      src/nextcloud/utilities/*: utilities/
+    plugin: dump
+    source: src/nextcloud/
+    organize:
+      config/: htdocs/config/
 
   # Download the boost headers for MySQL. Note that the version used may need to
   # be updated if the version of MySQL changes.
   boost:
-    plugin: copy
+    plugin: dump
     source: https://github.com/kyrofa/boost_tarball/raw/master/boost_1_59_0.tar.gz
     source-checksum: sha1/5123209db194d66d69a9cfa5af8ff473d5941d97
-    files:
-      '*': boost/
+    source-subdir: boost/
+    organize:
+      '*': boost/boost/
```

Comment:
Blech, looks like you hit a gross bug. Rather than force things to continue working this way, I suggest we remove the `source-subdir` and `organize` here completely, and simply change mysql to look for it straight in the staging area by updating [this line](https://github.com/nextcloud/nextcloud-snap/pull/466/files#diff-5fde7a6d86053f0e1d88c0a2a238941fR249) to use `$SNAPCRAFT_STAGE` instead of `$SNAPCRAFT_STAGE/boost`.

Reply:
No... scratch that, then files start interfering. Leave this the way it is.

Reply:
Now that you went over this again: The way it is now, snapcraft fails to do a rebuild without a full clean. If I go `snapcraft clean --step stage` afterwards and build again, it fails with some `boost/boost/boost/boost/boost/ recursion limit`.
Do you want me to open a bug on Launchpad for this? Otherwise, if you give me a bug number, I might be able to add my observations.

Reply:
We should be fine with:
```
source-subdir: boost/
organize:
  '*': boost/
```
Or drop `source-subdir` and ignore all files in the archive except `boost/`.

Both approaches would need the WITH_BOOST build parameter to be updated, as you mentioned.

Another simple way would be to call the target directory for the boost libs anything but `boost/`, e.g. `mysql_boost`, this should allow for saner approaches when organizing.

Reply:
What... really? You should log a bug, actually, this is dumb. I think I like your last idea best-- let's work around this "folder is the same name is the same name" thing and just call it "mysql_boost".

Reply:
But please add a comment in the YAML saying why.